transaction-logger/src/main.rs

```diff
@@ -0,0 +1,731 @@
+        let mut handles = Vec::with_capacity(parallel as usize);
+        for h in u64::from(*height)..u64::from(*height) + parallel {
+            let mut node = node.clone();
+            handles.push(async move {
+                let height: AbsoluteBlockHeight = h.into();
+                let blocks_at_height = node.get_blocks_at_height(height.into()).await?;
+                if blocks_at_height.len() != 1 {
+                    return Ok(Err(QueryFailure::NoBlocks));
+                }
+                let block = blocks_at_height[0];
+                let info = node.get_block_info(&block).await;
+                let info = match info {
+                    Ok(info) if !info.finalized => {
+                        return Ok(Err(QueryFailure::NotFinalized));
+                    }
+                    Ok(info) => info,
+                    Err(endpoints::QueryError::NotFound) => {
+                        return Ok(Err(QueryFailure::NotFound));
+                    }
+                    Err(endpoints::QueryError::RPCError(e)) => return Err(e.into()),
+                };
+                let summary = node.get_block_summary(&block).await?;
+                Ok::<_, anyhow::Error>(Ok((info, summary)))
+            });
+        }
+        let mut success = true;
+        for result in futures::future::join_all(handles).await {
+            match result? {
+                Ok((info, summary)) => {
+                    let mut with_addresses =
+                        Vec::with_capacity(summary.transaction_summaries.len());
+                    for summary in summary.transaction_summaries.into_iter() {
+                        let affected_addresses = summary.affected_addresses();
+                        let mut addresses = Vec::with_capacity(affected_addresses.len());
+                        // resolve canonical addresses. This part is only needed because the index
+                        // is currently expected by "canonical address",
+                        // which is only possible to resolve by querying the node.
+                        let mut seen = HashSet::with_capacity(affected_addresses.len());
+                        for address in affected_addresses.into_iter() {
+                            if let Some(addr) = canonical_cache.get(address.as_ref()) {
+                                let addr = AccountAddress::from(*addr);
+                                if !seen.insert(addr) {
+                                    addresses.push(addr);
+                                }
+                            } else {
+                                let ainfo = node
+                                    .get_account_info(address, &info.block_hash)
+                                    .await
+                                    .context("Error querying account info.")?;
+                                let addr = ainfo.account_address();
+                                log::debug!("Discovered new address {}", addr);
```

Comment:
To be precise I think this should be inside the "if seen". And it would also be useful to log if it was actually seen.

Reply:
Agree.